x-pack/plugins/features/server/feature_kibana_privileges.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/**
+ * Feature privilege definition
+ */
+export interface FeatureKibanaPrivileges {
+  /**
+   * Whether or not this specific privilege should be excluded from the base privileges.
+   */
+  excludeFromBasePrivileges?: boolean;
+
+  /**
+   * If this feature includes management sections, you can specify them here to control visibility of those
+   * pages based on user privileges.
+   *
+   * Example:
+   * // Enables access to the "Advanced Settings" management page within the Kibana section
+   * management: {
+   *   kibana: ['settings']
+   * }
+   */
+  management?: {
+    [sectionId: string]: string[];
+  };
+
+  /**
+   * If this feature includes a catalogue entry, you can specify them here to control visibility based on user permissions.
+   */
+  catalogue?: string[];
+
+  /**
+   * If your feature includes server-side APIs, you can tag those routes to secure access based on user permissions.
+   *
+   * Example:
+   * // Configure your routes with a tag starting with the 'access:' prefix
+   * server.route({
+   *   path: '/api/my-route',
+   *   method: 'GET',
+   *   handler: () => { ...},
+   *   options: {
+   *     tags: ['access:my_feature-admin']
+   *   }
+   * });
```

Comment:
nit: Can we use the `@example` TSDoc directive? Also applies to other examples in this file.

Reply:
Sure, I can add these

Reply:
Done